Makefile

```diff
@@ -112,16 +112,31 @@ man: qm.8.md ##             - Generates the QM man page
 	go-md2man --in qm.8.md --out qm.8
 
 .PHONY: dist
-dist: ##             - Creates the QM distribution package
-	tar cvz \
-		--exclude='.git' \
+DIST_FILE := /tmp/v$(VERSION).tar.gz
+TARGET_DIR := ./rpm
+EXCLUDE_DIRS := --exclude='.git' --exclude='.gitignore' --exclude='demos' --exclude='.github'
+
+# Autodetect SOURCE_DIR
+# Auto-detect SOURCE_DIR with version or default pattern
+SOURCE_DIR := $(shell \
+	[ -d "../qm-$(VERSION)" ] && echo "../qm-$(VERSION)" || \
+	[ -d "../qm" ] && echo "../qm" || \
+	echo "")
+
+dist: ## Creates the QM distribution package
+	@if [ -z "$(SOURCE_DIR)" ]; then \
+		echo "Error: qm directory not found."; \
```

Comment:
You are failing here 
```
2024-10-28 06:27:20.116 logging.py        INFO   Error: qm directory not found.
```

Better to work with static directory under /tmp dir
Try to print the SOURCE_DIR

Reply:
yep, I was trying to avoid static /tmp but I agree leet's make it happen this way. We can always improve the code with more people helping in a long run.